crates/uv-build/src/lib.rs

```diff
@@ -849,20 +867,48 @@ async fn run_python_script(
     venv: &Virtualenv,
     script: &str,
     source_tree: &Path,
+    mut environment_variables: FxHashMap<String, String>,
 ) -> Result<Output, Error> {
+    // First check user supplied environment variables
+    let path = environment_variables.remove("PATH");
+    let os_path = env::var_os("PATH");
```

Comment:
Shouldn't the caller determine the entire `PATH`? Then we also wouldn't need to modify it and could pass it in as a reference.

Reply:
Sure and you want to get the env var at the caller site as well?

Reply:
I hope i got this in the right order, but i'd try to get `PATH` from `environment_variables` first and then from the real env, prepend the build venv and then pass to the command so that our now computed `PATH` takes precedence over `.envs(environment_variables)`

Reply:
So I've now changed this to kreep track of the path seperately, so they are computed one and passed in by reference.